### PR TITLE
Foundational Security: ELB

### PR DESCRIPTION
**Pull request by disq**
~Waiting for ACM to really implement ELB.2~

ACM PR here https://github.com/cloudquery/cq-provider-aws/pull/313

Still not sure about this rule:
> To create a certificate, you can use either ACM or a tool that supports the SSL and TLS protocols, such as OpenSSL. Security Hub recommends that you use ACM to create or import certificates for your load balancer.

My understanding is, it's allowed to use an "imported" (not Amazon issued) cert in the ACM as long as it's in the ACM (is there another way of setting cert to a LB anyway?)

So in the [query](https://github.com/cloudquery-policies/aws/blob/foundational_security_elb/queries/elbv1/cert_provided_by_acm.sql) I `LEFT JOIN` with `aws_acm_certificates` and see if the `listener_ssl_certificate_id` matches the `arn` or not. Hope it's the way to do it

### CODE REVIEW

File: queries/elbv2/alb_drop_http_headers.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select lb.account_id, lb.region, lb.name from aws_elbv2_load_balancers lb
+join aws_elbv2_load_balancer_attributes a on a.load_balancer_cq_id=lb.cq_id
+where lb.type='application' AND a.routing_http_drop_invalid_header_fields != true;
```
Comment (yevgenypats) [inline queries/elbv2/alb_drop_http_headers.sql:3]: better to use `is not true`

File: queries/elbv2/alb_deletion_protection_enabled.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+select lb.account_id, lb.region, lb.name from aws_elbv2_load_balancers lb
+join aws_elbv2_load_balancer_attributes a on a.load_balancer_cq_id=lb.cq_id
+where lb.type='application' AND a.deletion_protection != true;
```
Comment (yevgenypats) [inline queries/elbv2/alb_deletion_protection_enabled.sql:3]: better to use `is not true`

**Review by yevgenypats — CHANGES_REQUESTED** [review]
better to use `is not true` or `is distinct from` for non-boolean. There was a slack discussion around it but basically if you have null values it will fail:

```
postgres=# select 1 where null != true;
 ?column? 
----------
(0 rows)
```

**Review by yevgenypats — APPROVED** [review]

File: foundational_security/policy.hcl [hunk 3]
```diff
@@ -275,37 +275,37 @@ policy "foundational_security" {
...
         query "4" {
             description = "Application load balancers should be configured to drop HTTP headers"
-            query = "select 1;"
+            query = file("queries/elbv2/alb_drop_http_headers.sql")
         }
 
         query "5" {
             description = "Application and Classic Load Balancers logging should be enabled"
-            query = "select 1;"
+            query = file("queries/elbv2/alb_logging_enabled.sql")
```
Comment (bbernays) [inline foundational_security/policy.hcl:293]: Do we need to break elbv2 and elbv1 into different folders? It seems weird for some queries like this one that check both

Comment (disq) [reply foundational_security/policy.hcl:293]: yeah, I don't know, I kind of agree. just move all to `elb/`?

Comment (bbernays) [reply foundational_security/policy.hcl:293]: I think that is a good idea

Comment (disq) [reply foundational_security/policy.hcl:293]: then if something just applies to the classic load balancer we can prefix that file with `elc`?

Comment (disq) [reply foundational_security/policy.hcl:293]: @yevgenypats what do you think?

Comment (yevgenypats) [reply foundational_security/policy.hcl:293]: yeah sounds good. let's do it.

**Review by spangenberg — APPROVED** [review]
LGTM

**Pull request merged**